src/uwtools/drivers/forecast.py

```diff
@@ -40,33 +50,87 @@ def validate(self):
 
-        ''' Holds the knowledge for how to modify a list of fix files and 
-        stages them in the working directory. Likely gets all its info from 
-        config_obj. Calls data mover tool (could be python copy). Fix files 
+        ''' Holds the knowledge for how to modify a list of fix files and
+        stages them in the working directory. Likely gets all its info from
+        config_obj. Calls data mover tool (could be python copy). Fix files
         usually are specific to a given named grid and resolution. '''
 
     def create_namelist(self):
-        ''' Collects all the user inputs required to create a namelist 
+        ''' Collects all the user inputs required to create a namelist
         file, calling the existing namelist config tools. For SRW,
         this will take the fix files and model_config previously created to
         to generate config.yaml'''
 
+    def create_directory_structure(self, run_directory, exist_act="delete"):
+        ''' Collects the name of the desired run directory, and has an
+        optional flag for what to do if the run directory specified already
+        exists. Creates the run directory and adds subdirectories
+        INPUT and RESTART. Verifies creation of all directories.
+
+        Args:
+           run_directory: path of desired run directory
+           exist_act: - could be any of 'delete', 'rename', 'quit'
+                      - how program should act if run directory exists
+                      - default is to delete old run directory
+           Returns: None
+        '''
+
+        # Caller can only provide correct argument
+        if exist_act not in ["delete", "rename", "quit"]:
+            raise ValueError("Bad argument to create_directory_structure")
+
+        # Exit program with error if caller chooses to Quit
+        if exist_act == "quit":
+            logging.critical("User chose Quit option")
+            sys.exit(1)
+
+        # Try to delete existing run directory if option is Delete
+        try:
+            if exist_act == "delete" and os.path.isdir(run_directory):
+                shutil.rmtree(run_directory)
+        except (RuntimeError, FileExistsError) as del_error:
+            raise RuntimeError("Could not delete " +
+                               "old run directory") from del_error
+
+        # Try to rename existing run directory if caller chooses Rename
+        try:
+            if exist_act == "rename" and os.path.isdir(run_directory):
+                now = datetime.now()
+                save_dir = run_directory + now.strftime("_%Y%m%d_%H%M%S")
+                shutil.move(run_directory, save_dir)
+        except (RuntimeError, FileExistsError) as rename_error:
+            raise RuntimeError("Could not rename " +
+                               "old run directory") from rename_error
+
+        # Create new run directory with two required subdirectories
+        try:
+            os.makedirs(run_directory + "/INPUT")
```

Comment:
We can achieve more platform independence if we try to use `os.path.join()` when we build up paths. It is a bit annoying that it's so much more syntax, though.

Reply:
Changed the three instances